modules/cassandra/cassandra.go

```diff
@@ -12,20 +15,34 @@ import (
 )
 
 const (
-	port = "9042/tcp"
+	port    = "9042/tcp"
+	sslPort = "9142/tcp"
 )
 
+//go:embed testdata/cassandra-ssl.yaml
+var sslConfigYAML []byte
+
 // CassandraContainer represents the Cassandra container type used in the module
 type CassandraContainer struct {
 	testcontainers.Container
+	settings options
 }
 
 // ConnectionHost returns the host and port of the cassandra container, using the default, native 9042 port, and
 // obtaining the host and exposed port from the container
 func (c *CassandraContainer) ConnectionHost(ctx context.Context) (string, error) {
+	if c.settings.tlsEnabled {
+		return c.PortEndpoint(ctx, sslPort, "")
+	}
 	return c.PortEndpoint(ctx, port, "")
 }
 
+// TLSConfig returns the TLS configuration for secure client connections.
+// Returns nil if TLS is not enabled on the container.
+func (c *CassandraContainer) TLSConfig() *tls.Config {
```

Comment:
bug: let's document the new container method in the `docs/modules/cassandra.md` file. Please remember to use the marker as described in the contributing docs: https://golang.testcontainers.org/contributing/#adding-functional-options

It could be of interest to embed a code snippet linking to `// TLSConfig {` (see https://golang.testcontainers.org/contributing/#adding-code-snippets)